### PR TITLE
adjust collection member solr query to correctly return collection members

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Fixes sul-dlss/argo#2243

Basically, the collection member solr query wasn't returning anything, preventing collection member release from working correctly.

## How was this change tested?

Tested on -stage, by running a call from argo-stage.

### CODE REVIEW

File: app/controllers/members_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class MembersController < ApplicationController
   # Return the published members of this collection
   def index
     solr_params = {
-      q: "is_member_of_collection_ssim:\"#{params[:id]}\" published_dttsim:[* TO *]",
+      q: "is_member_of_collection_ssim:\"info:fedora/#{params[:id]}\" published_dttsim:[* TO *]",
```
Comment (jcoyne) [inline app/controllers/members_controller.rb:8]: How about: `ActiveFedora::Base.internal_uri(params[:id])` Even better might be to do:
```
_query_:"{!raw f=is_member_of_collection_ssim}#{ActiveFedora::Base.internal_uri(params[:id])}"
```

Comment (peetucket) [reply app/controllers/members_controller.rb:8]: Didn't realize there was an internal_uri method, that seems like a natural change, but I'm not sure I get that `!raw` syntax.  How does that incorporate the other parts of the query?

Comment (jcoyne) [reply app/controllers/members_controller.rb:8]: The raw query parser does no analysis or transformation: https://lucene.apache.org/solr/guide/6_6/other-parsers.html#OtherParsers-RawQueryParser

`_query_` is a magic field for the dismax parser: https://lucene.apache.org/solr/guide/6_6/the-extended-dismax-query-parser.html

Comment (peetucket) [reply app/controllers/members_controller.rb]: I just used `ActiveFedora::Base.internal_uri(params[:id])` but kept the query syntax - its not clear to me what the advantage is of using the `_query_` magic field?

Comment (jcoyne) [reply app/controllers/members_controller.rb]: If you don't use the `_query_` field you can't escape to use the raw parser, you'll use the default dismax parser.

File: app/controllers/members_controller.rb [hunk 2]
```diff
@@ -4,8 +4,9 @@
 class MembersController < ApplicationController
   # Return the published members of this collection
   def index
+    pid = params[:object_id] || params[:id]
```
Comment (jcoyne) [inline app/controllers/members_controller.rb:7]: Why do we need to look at both of these?

Comment (jcoyne) [reply app/controllers/members_controller.rb:7]: It looks to me like `object_id` is the only valid choice:
```
$ bin/rails routes -g members
        Prefix Verb URI Pattern                              Controller#Action
object_members GET  /v1/objects/:object_id/members(.:format) members#index {:format=>:json}
```

**Review by jcoyne — CHANGES_REQUESTED** [review]
It looks to me like `spec/requests/members_for_collection_spec.rb` is missing an assertion. Can you add it please?

File: app/controllers/members_controller.rb (hunk 2 above)
Comment (peetucket) [reply app/controllers/members_controller.rb:7]: ah, when looking for that and missed it, but noticed it wasn't working with params[:id], fixed it by changing to params[:object_id] (and it works now)

**Comment by peetucket** [comment]
> It looks to me like `spec/requests/members_for_collection_spec.rb` is missing an assertion. Can you add it please?

what assertion are you thinking of?

**Comment by jcoyne** [comment]
The one that says we call solr with the correct query (contains the id).

**Comment by peetucket** [comment]
adjusted the member spec to confirm correct solr params are sent

**Review by jcoyne — APPROVED** [review]

**Pull request merged**